### DISCUSSION TITLE
Format option for components

### DISCUSSION THREAD

**Discussion by FarrunDragon** [FakeFishGames/Barotrauma#8699] [Ideas and suggestions]
I think it would provide a lot of easier use for outgoing signals if there are formatting options (or even a regex field) for the components. This way you could limit digits, prepend formats (without needing the concat component), or other formats.

**Comment by James-E-A** [comment]
A simple change to the regex component could do this, I think.

Currently, the "Use capture groups" setting does _not_ work like a normal regex library -- if I have the regex `^power (\d+)$`, then the Output field [_should_ take `$1` to output the captured number](https://docs.microsoft.com/en-us/visualstudio/ide/using-regular-expressions-in-visual-studio?view=vs-2022#capture-groups-and-replacement-patterns) -- but it instead expects `1`.

If the regex output field were changed to accept a string, with substitution groups injected via dollar-sign, it would instantly become far more versatile (and the capture-group checkbox would be rendered unnecessary, too).
